src/libstore/derivations.hh

```diff
@@ -342,16 +346,16 @@ struct Derivation : BasicDerivation
      * 2. Input placeholders are replaced with realized input store
      *    paths.
      */
-    std::optional<BasicDerivation> tryResolve(Store & store) const;
+    std::optional<BasicDerivation> tryResolve(Store & store, Store * evalStore = nullptr) const;
```

Comment:
(nit): I guess the reason `evalStore` is a pointer and not a reference is that it's optional. Would it make sense to have either:

```suggestion
    std::optional<BasicDerivation> tryResolve(Store & store, Store & evalStore store) const;
```

(if C++ allows it, which I think it doesn't)

or

```suggestion
    std::optional<BasicDerivation> tryResolve(Store & store, Store & evalStore) const;
    std::optional<BasicDerivation> tryResolve(Store & store) const
    { return tryResolve(store, store); }
```

Reply:
I agree this is better, but we do it the `evalStore = nullptr` many other places now, so I think it is better to revisit that separtely:

```bash
$ git grep -c 'evalStore = nullptr' master
upstream/master:src/libstore/build/local-derivation-goal.cc:2
upstream/master:src/libstore/remote-store.hh:1
upstream/master:src/libstore/store-api.hh:5
```

Reply:
Makes sense. Let's keep it for now